src/auth/environment.ts

```diff
@@ -0,0 +1,130 @@
+        await this.auth.tryAutoConnect()
+
+        const conn = isIamConnection(this.auth.activeConnection)
+            ? this.auth.activeConnection
+            : await promptForConnection(this.auth, 'iam')
+
+        // User selected to edit credentials or add a new connection
+        // We're not within a command, otherwise we'd throw an error here to signal cancellation
+        if (!conn) {
+            return
+        }
+
+        if (!isIamConnection(conn)) {
+            throw new ToolkitError('No valid AWS IAM connection found.', { code: 'NoConnection' })
+        }
+
+        const validatedConn =
+            this.auth.getConnectionState(conn) !== 'valid'
+                ? ((await this.auth.reauthenticate(conn)) as unknown as IamConnection)
+                : conn
+
+        const { env } = await injectCredentials(this.auth, validatedConn, 'TerminalProfile')
+
+        return new vscode.TerminalProfile({
+            env,
+            strictEnv: true,
+            name: `AWS (${validatedConn.label})`,
+            message: `Using AWS connection "${validatedConn.label}"`,
+            isTransient: true,
+        } as vscode.TerminalOptions)
+    }
+
+    public dispose(): void {
+        vscode.Disposable.from(...this.#disposables).dispose()
+        this.collection.clear()
+        clearTimeout(this.#expirationTimer)
+    }
+
+    private async handleUpdate(conn: Connection | undefined = this.auth.activeConnection) {
+        // This will not work well with multiple users of `EnvironmentVariableCollection`
+        this.collection.clear()
+        clearTimeout(this.#expirationTimer)
+
+        if (!this.enabled) {
+            return
+        }
+
+        if (conn && this.auth.getConnectionState(conn) === 'valid' && isIamConnection(conn)) {
+            await this.updateCollection(conn)
+        }
+    }
+
+    private async updateCollection(conn: IamConnection): Promise<void> {
+        const { expiration, env } = await injectCredentials(this.auth, conn, 'AutomaticInjection', {})
+        if (expiration) {
+            this.#expirationTimer = +setTimeout(() => this.handleUpdate(conn), expiration.getTime() - Date.now())
+        }
+
+        for (const [k, v] of Object.entries(env)) {
+            if (v !== undefined) {
+                this.collection.replace(k, v)
+            } else {
+                this.collection.delete(k)
+            }
+        }
+    }
+}
+
+export async function injectCredentials(
+    auth: Auth,
+    connection: IamConnection,
+    source?: 'TerminalProfile' | 'AutomaticInjection',
+    env = process.env
+) {
+    return telemetry.aws_injectCredentials.run(async () => {
+        telemetry.record({ source, passive: true })
+
+        const credentials = await connection.getCredentials()
+
+        return {
+            env: {
+                ...env,
+                AWS_REGION: await auth.getDefaultRegion(connection),
```

Comment:
I wonder if we should use guessDefaultRegion here. https://github.com/aws/aws-toolkit-vscode/blob/be37644158cc5b9f0eb9f33dc01eed49d02ca5aa/src/shared/regions/regionProvider.ts#L97